CONVENTIONS.md

```diff
@@ -0,0 +1,55 @@
+# Coding Conventions
+
+The following document outlines a series of conventions that must be used for all code contributions to Fabric API, and recommendations for other modding projects.
+
+## General
+
+All code should meet the Yarn naming conventions which can be found [here](https://github.com/FabricMC/yarn/blob/20w21a/CONVENTIONS.md).
+
+All calls of associated methods and uses of class fields should be prefixed with `this.` For example, setting the field `name` on a class should always be done by calling `this.name = "name"` rather than `name = "name"`. This is especially pertinent in constructors where constructor parameters often share a name with class fields.
+
+## Annotations
+
+All methods that can potentially return a null value should be annotated in-line using the JetBrains `@Nullable` right before the return type definition for the method. For example, a method named `possiblyGetName` that can potentially return a null value should be written as `public @Nullable String possiblyGetName()...`.
+
+Likewise all methods that can accept a null value should be annotated using the JetBrains `@Nullable` right before the method parameter. For example, a method named `food` that can accept a null value should be written as follows `public Item.Settings food(@Nullable FoodComponent food)...`.
+
+## Flow Control
+
+Wherever it makes sense the ternary operator ought to be preferred over an if statement for the sake of brevity, but never at the expense of readability. If the statement contains a long if condition it should not be turned into a ternary operator because the line will be too long and the statement will lose readability. If the statement contains a short if else statement and can be expressed simply as a ternary operator, that ought to be preferred.
+
+## Enums
+
+Enum values that are referenced internally (values referenced within methods associated with their own enum definition) must always use `Enum.VALUE` never simply `VALUE`.
```

Comment:
I am arguably in favor of requiring that you qualify references to enum members, but this is should probably be part of a future pull request for actual conventions.